plugins/woocommerce/src/Blocks/Utils/BlocksSharedState.php

```diff
@@ -0,0 +1,152 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Automattic\WooCommerce\Blocks\Utils;
+
+use InvalidArgumentException;
+
+/**
+ * Manages the registration of interactivity config and state that is commonly shared by WooCommerce blocks.
+ * Initialization only happens on the first call to initialize_shared_config.
+ * Intended to be used as a singleton.
+ */
+class BlocksSharedState {
+
+	/**
+	 * The consent statement for using private APIs of this class.
+	 *
+	 * @var string
+	 */
+	private $consent_statement = 'I acknowledge that using private APIs means my theme or plugin will inevitably break in the next version of WooCommerce';
```

Comment:
@luisherranz I've taken the same approach as private APIs from Gutenberg, and required passing a consent string to use either of these methods. This should allow us not to worry about deprecating these methods or their side effects in future versions.

I don't expect 3PDs to use these methods but this does actively discourage them from doing so.

I've also consolidated the cart state here as well.